WordPress/gutenberg
# Block library: Patterns API integration with Media & Text block

## Description
Builds on top of #18283.

Part 3.a of #16283.

This PR explores a possible way to integrate Patterns API with the existing Media & Text block as proposed in in https://github.com/WordPress/gutenberg/issues/16283#issuecomment-508132004

> <img width="800" alt="Example - Media Text" src="https://user-images.githubusercontent.com/1202812/60599876-30666300-9d7d-11e9-9cf2-ec5cc3d02e9b.png">

As part of the implementation process, I reorganized the folder structure of the Media & Text block to group together all files related to the `edit` component.

## How has this been tested?
See the recorder screencast.

1. Open a new post.
2. Insert Media & Text block.
3. Pick one of the patterns and make sure it gets applied.

## Screenshots 

![Screen Shot 2019-11-06 at 19 37 01](https://user-images.githubusercontent.com/699132/68327009-da0f8500-00cc-11ea-8c6e-d041a0a0d418.png)

![media-text](https://user-images.githubusercontent.com/699132/68463613-55794f80-020f-11ea-9c04-a8361e3c51d2.gif)


## Types of changes





## Checklist:
